name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/ibflex/IBFlexStatementExtractor.java

```diff
@@ -428,6 +430,20 @@ else if (element.hasAttribute("dateTime"))
             String targetCurrency = parts[0];
             String sourceAmount = element.getAttribute("tradeMoney");
             String targetAmount = element.getAttribute("quantity");
+            BigDecimal exachangeRate = asExchangeRate(element.getAttribute("tradePrice"));
+            // If "quantity" is negative, then we're dealing with pair like
+            // USD.CAD, exchanging USD to CAD, or maybe EUR.USD, exchanging
+            // EUR to USD.
+            if (targetAmount.startsWith("-"))
```

Comment:
Can you contribute an example where the targetAmount does not start with a "-"?

Because all examples we have is buySell="SELL". 
And then it sounds to me that is the more reasonable attribute to use to determine the source and target pair.
If it is "SELL", and the symbol is "EUR.USD" then it is EUR is source and USD is target.
If it is "BUY" (but I do not have an example, and the symbol is "EUR.USD", then USD is the source and EUR the target.

Reply:
> buySell=
> And then it sounds to me that is the more reasonable attribute to use to determine the source and target pair.

I can confirm that this will work too. Preparing example may take some time though.